### PR TITLE
change ``np.ones`` patterns to ``np.full``

### LINKED ISSUE #165: Use `np.full`?
In several pieces of code, an array of a constant value is created by something like

```py
Q = np.ones((num_nodes, num_nodes))
Q = Q * np.inf
```

This can be simplified with [`np.full`](https://numpy.org/doc/stable/reference/generated/numpy.full.html):

```py
Q = np.full((num_nodes, num_nodes), np.inf)
```

### PR DESCRIPTION
**Pull request by shlff**
Good morning @jstac and @oyamad , this PR fix #165 by replacing all ``np.ones`` patterns with ``np.full`` in ``lecture-python.myst``, suggested by @oyamad .

### CODE REVIEW

**Comment by jstac** [comment]
Hi @mmcky , would you mind to review this?  I'm not sure why the build is failing.

**Comment by mmcky** [comment]
@jstac the lecture `career.md` is failing during the execution tests. I will take a look this afternoon but one of the changes to the code must not be working properly.

**Comment by mmcky** [comment]
@shlff there seems to be an issue with `njit` in the `career` lecture. I don't have time this afternoon to fix this and I will be travelling tomorrow. Would you mind to look into this?

[career.ipynb.zip](https://github.com/QuantEcon/lecture-python.myst/files/6796253/career.ipynb.zip)

```bash
---------------------------------------------------------------------------
TypingError                               Traceback (most recent call last)
<ipython-input-7-a1d4eb67d899> in <module>
      1 cw = CareerWorkerProblem()
      2 T, get_greedy = operator_factory(cw)
----> 3 v_star = solve_model(cw, verbose=False)
      4 greedy_star = get_greedy(v_star)
      5 

<ipython-input-6-aefaf00ce40e> in solve_model(cw, use_parallel, tol, max_iter, verbose, print_skip)
     14 
     15     while i < max_iter and error > tol:
---> 16         v_new = T(v)
     17         error = np.max(np.abs(v - v_new))
     18         i += 1

~/anaconda3/envs/quantecon/lib/python3.8/site-packages/numba/core/dispatcher.py in _compile_for_args(self, *args, **kws)
    413                 e.patch_message(msg)
    414 
--> 415             error_rewrite(e, 'typing')
    416         except errors.UnsupportedError as e:
    417             # Something unsupported is present in the user code, add help info

~/anaconda3/envs/quantecon/lib/python3.8/site-packages/numba/core/dispatcher.py in error_rewrite(e, issue_type)
    356                 raise e
    357             else:
--> 358                 reraise(type(e), e, None)
    359 
    360         argtypes = []

~/anaconda3/envs/quantecon/lib/python3.8/site-packages/numba/core/utils.py in reraise(tp, value, tb)
     78         value = tp()
     79     if value.__traceback__ is not tb:
---> 80         raise value.with_traceback(tb)
     81     raise value
     82 

TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function matmul>) found for signature:
 
 >>> matmul(array(float64, 1d, C), array(int64, 2d, C))
 
There are 2 candidate implementations:
  - Of which 2 did not match due to:
  Overload in function 'MatMul.generic': File: numba/core/typing/npydecl.py: Line 987.
    With argument(s): '(array(float64, 1d, C), array(int64, 2d, C))':
   Rejected as the implementation raised a specific error:
     TypingError: '@' arguments must all have the same dtype
  raised from /Users/matthewmckay/anaconda3/envs/quantecon/lib/python3.8/site-packages/numba/core/typing/npydecl.py:932

During: typing of intrinsic-call at <ipython-input-5-491a1abd86c9> (24)

File "<ipython-input-5-491a1abd86c9>", line 24:
    def T(v):
        <source elided>
                v2 = θ[i] + G_mean + β * v[i, :] @ G_probs        # New job
                v3 = G_mean + F_mean + β * F_probs @ v @ G_probs  # New life
```

File: lectures/career.md [hunk 1]
```diff
@@ -288,7 +288,7 @@ def solve_model(cw,
     T, _ = operator_factory(cw, parallel_flag=use_parallel)
 
     # Set up loop
-    v = np.ones((cw.grid_size, cw.grid_size)) * 100  # Initial guess
+    v = np.full((cw.grid_size, cw.grid_size), 100)  # Initial guess
```
Comment (oyamad) [inline lectures/career.md]: Maybe the value should be `float`?

```suggestion
    v = np.full((cw.grid_size, cw.grid_size), 100.)  # Initial guess
```

Comment (mmcky) [reply lectures/career.md]: nice thinking @oyamad

**Comment by mmcky** [comment]
nice work @oyamad. @jstac @shlff this is now building with all lectures executing.

**Review by mmcky — APPROVED** [review]
thanks to @oyamad suggested the lectures are now executing. 

cc @jstac

**Comment by jstac** [comment]
Thanks all, much appreciated.

**Pull request merged**